src/lib.rs

```diff
@@ -184,6 +185,17 @@ fn is_isomorphic(
     Ok(res)
 }
 
+#[pyfunction]
+#[text_signature = "(first, second, /)"]
+fn combine_graphs(
```

Comment:
I'm not sure about this function name, since based on my reading of the code it will take the 2 graphs and replace the nodes from first if there is a node from second with the same weight. Otherwise the node from second graph is added as a disjoint node. Maybe `merge_digraphs` or `combine_digraphs_by_weight`? I just want to try to avoid confusion with other methods of combination, like what we're adding in: https://github.com/Qiskit/retworkx/pull/66